tfexec/cmd.go

```diff
@@ -193,6 +195,32 @@ func (tf *Terraform) buildTerraformCmd(ctx context.Context, mergeEnv map[string]
 	return cmd
 }
 
+func (tf *Terraform) buildTerraformCmdWithGracefulShutdown(ctx context.Context, gracefulShutdownPeriod time.Duration, mergeEnv map[string]string, args ...string) *exec.Cmd {
+	cmd := exec.CommandContext(ctx, tf.execPath, args...)
+
+	cmd.Env = tf.buildEnv(mergeEnv)
+	cmd.Dir = tf.workingDir
+
+	if gracefulShutdownPeriod > 0 {
+		tf.logger.Printf("[INFO] configuring gracefully cancellation for command: %s", cmd.String())
+		cmd.WaitDelay = gracefulShutdownPeriod
```

Comment:
question: If the terraform process doesn't exit in time after sending `os.Interrupt`. Is this case exist? where do we handle it?

Reply:
Then it will be force killed and we will still have a chance to experience data corruption.

Reply:
We can also add an option to set `cmd.WaitDelay` to zero, then the program will wait indefinitely until the subprocess finishes and close its pipe.

Let me know what you think.

Reply:
I updated the API to allow zero `cmd.WaitDelay`. PTAL